### PR TITLE
[docs][api] Notes on ycql transaction api

### PR DESCRIPTION
**Pull request by ddhodge**
Notes on YCQL transaction statement

@netlify /preview/api/ycql/dml_transaction/

### CODE REVIEW

File: docs/content/preview/api/ycql/dml_transaction.md [hunk 1]
```diff
@@ -32,13 +32,14 @@ transaction_block ::= BEGIN TRANSACTION
                       END TRANSACTION ';'
 ```
 
-Where
+Where `insert`, `update`, and `delete` are [INSERT](../dml_insert), [UPDATE](../dml_update/), and [DELETE](../dml_delete/) statements.
 
-- `insert`, `update`, and `delete` are [INSERT](../dml_insert), [UPDATE](../dml_update/), and [DELETE](../dml_delete/) statements.
+- When using `BEGIN TRANSACTION`, you don't use a semicolon.
```
Comment (sumitdatabase) [inline docs/content/preview/api/ycql/dml_transaction.md:37]: May be one line for syntax with 'BEGIN TRANSACTION' and another line for syntax with 'START TRANSACTION'? Like

L 37 --> When using `BEGIN TRANSACTION`, you don't use a semicolon. End the transaction block with `END TRANSACTION ;`

L38 --> When using `START TRANSACTION ;`, End the transaction block with `COMMIT ;`

File: docs/content/preview/api/ycql/dml_transaction.md [hunk 2]
```diff
@@ -51,6 +51,9 @@ transaction_block ::= START TRANSACTION ';'
 
 - An error is raised if transactions are not enabled in any of the tables inserted, updated, or deleted.
 - Currently, an error is raised if any of the `INSERT`, `UPDATE`, or `DELETE` statements contains an `IF` clause.
+- If transactions are enabled for a table, its indexes must have them enabled as well, and vice versa.
+- There is no explicit rollback.
+- DDLs are always committed immediately, even if they are inside a transaction block.
```
Comment (sumitdatabase) [inline docs/content/preview/api/ycql/dml_transaction.md:60]: May be a bit more clarity like this?

"..even if they are inside a transaction block and the transaction is rolled back, by disconnecting the session."

**Review by sumitdatabase — COMMENTED** [review]
Thanks @ddhodge for the changes and getting it so quick. The changes to the file docs/content/preview/api/ycql/dml_transaction.md all looks good to me.

File: docs/content/preview/api/ycql/_index.md [hunk 3]
```diff
@@ -72,7 +72,7 @@ Statement | Description |
 [`GRANT ROLE`](ddl_grant_role) | Grant a role to another role |
 [`REVOKE ROLE`](ddl_revoke_role) | Revoke a role from another role |
 
-- The following table lists all security statements that are not yet implemented.
+The following table lists all security statements that are not yet implemented.
 
 Primitive Type |
```
Comment (polarweasel) [inline docs/content/preview/api/ycql/_index.md:77]:
```suggestion
Statement |
```

File: docs/content/preview/api/ycql/dml_transaction.md [hunk 4]
```diff
@@ -32,13 +32,14 @@ transaction_block ::= BEGIN TRANSACTION
                       END TRANSACTION ';'
 ```
 
-Where
+Where `insert`, `update`, and `delete` are [INSERT](../dml_insert), [UPDATE](../dml_update/), and [DELETE](../dml_delete/) statements.
 
-- `insert`, `update`, and `delete` are [INSERT](../dml_insert), [UPDATE](../dml_update/), and [DELETE](../dml_delete/) statements.
+- When using `BEGIN TRANSACTION`, you don't use a semicolon. End the transaction block with `END TRANSACTION ;`.
```
Comment (polarweasel) [inline docs/content/preview/api/ycql/dml_transaction.md:37]: Do we want to take a belt-and-braces approach to this one? If so...
```suggestion
- When using `BEGIN TRANSACTION`, you don't use a semicolon. End the transaction block with `END TRANSACTION ;` (with a semicolon).
```

File: docs/content/preview/api/ycql/dml_transaction.md [hunk 5]
```diff
@@ -32,13 +32,14 @@ transaction_block ::= BEGIN TRANSACTION
...
+- There is no `COMMIT` for transactions started using `BEGIN`.
 
-### ANSI SQL syntax
+### SQL syntax
 
-Alternatively, YugabyteDB supports ANSI SQL `START TRANSACTION` and `COMMIT` statements.
+YCQL also supports SQL `START TRANSACTION` and `COMMIT` statements.
 
 ```
```
Comment (polarweasel) [inline docs/content/preview/api/ycql/dml_transaction.md:44]: For glorious syntax highlighting, ebnf is the language for all these. :)

````suggestion
```ebnf
````

**Review by polarweasel — APPROVED** [review]
Couple little changes, then GTG

**Pull request merged**